agent/fsx/factory/factory.go

```
@@ -43,12 +46,24 @@ type fsxClientCreator struct{}
 
 func (*fsxClientCreator) NewFSxClient(region string,
 	creds credentials.IAMRoleCredentials) fsxclient.FSxClient {
-	cfg := aws.NewConfig().
-		WithHTTPClient(httpclient.New(roundtripTimeout, false, agentversion.String(), config.OSType)).
-		WithRegion(region).
-		WithCredentials(
-			awscreds.NewStaticCredentials(creds.AccessKeyID, creds.SecretAccessKey,
-				creds.SessionToken))
-	sess := session.Must(session.NewSession(cfg))
-	return fsx.New(sess)
+	// Create credentials provider for SDK v2
+	staticCreds := awscreds.NewStaticCredentialsProvider(
+		creds.AccessKeyID,
+		creds.SecretAccessKey,
+		creds.SessionToken,
+	)
+
+	cfg, err := awsconfig.LoadDefaultConfig(
+		context.TODO(),
+		awsconfig.WithRegion(region),
+		awsconfig.WithCredentialsProvider(staticCreds),
+		awsconfig.WithHTTPClient(httpclient.New(roundtripTimeout, false, agentversion.String(), config.OSType)),
+	)
+	if err != nil {
+		return nil
```

This should also return an error for the error case, right?

---

good catch! fixed it